source/rust_verify/example/invariants.rs

```diff
@@ -0,0 +1,169 @@
+// rust_verify/tests/example.rs expect-failures
```

Comment:
This may not make sense here (e.g. if it's important to showcase the errors), but consider moving the failing cases to separate tests: with `expect-failures` we're less likely to notice if the examples drift and become incorrect, becausse the unexpected errors will not cause a test failure.

Reply:
FWIW, all these examples are in the `tests/` directory as well, which is obviously more robust. I made the examples file here because it was easier to experiment & mess around.

Is there an easy way to run all the tests in a test file and get their output (even when the tests pass)? If so, it would be much easier to experiment in the main test file, instead. Then I'd leave the `examples` file for proper, working examples.

Reply:
I generally work in the test file, selecting a specific test with `tools/cargo.s test -p rust_verify --test invariants -- --nocapture substring_match_of_the_test_fn`. You can also ask for the full backtrace and to log IRs (which only makes sense for a single test), with `VERIFY_LOG_IR_PATH=log/ RUST_BACKTRACE=1 ./tools/cargo.sh test -p rust_verify --test refs -- --nocapture arg_spec` for example.

> Then I'd leave the examples file for proper, working examples.

This would be nice!